gef-ui/server/utils/helpers.test.js

```diff
@@ -756,9 +861,97 @@ describe('mapSummaryList()', () => {
         user: MOCK_REQUEST,
       };
 
-      const { text } = mapSummaryList(MOCK_UNISSUED_FACILITY, mockedDisplayItemsEnd, mapSummaryParams, true)[0].actions.items[0];
+      const actionItems = mapSummaryList(MOCK_UNISSUED_FACILITY, mockedDisplayItemsEnd, mapSummaryParams, true)[0].actions.items;
+      // should be allowed to change so should display change
+      expect(actionItems).toEqual([
+        {
+          attributes: {
+            'data-cy': `coverEndDate-action`,
+          },
+          classes: 'govuk-!-display-none',
+        },
+      ]);
+    });
+
+    it('isUsingFacilityEndDate', () => {
+      const mockedDisplayItemsEnd = [
+        {
+          label: 'Is using facility end date',
+          id: 'isUsingFacilityEndDate',
+        },
+      ];
+
+      const facility = { ...MOCK_UNISSUED_FACILITY, details: { ...MOCK_UNISSUED_FACILITY.details, isUsingFacilityEndDate: true } };
+
+      const mapSummaryParams = {
+        app: MOCK_AIN_APPLICATION,
+        user: MOCK_REQUEST,
+      };
+
+      const actionItems = mapSummaryList(facility, mockedDisplayItemsEnd, mapSummaryParams, true)[0].actions.items;
+      // should be allowed to change so should display change
+      expect(actionItems).toEqual([
+        {
+          attributes: {
+            'data-cy': `isUsingFacilityEndDate-action`,
+          },
+          classes: 'govuk-!-display-none',
+        },
+      ]);
+    });
+
+    it('bankReviewDate', () => {
+      const mockedDisplayItemsEnd = [
+        {
+          label: 'Bank review date',
+          id: 'bankReviewDate',
+        },
+      ];
+
+      const facility = { ...MOCK_UNISSUED_FACILITY, details: { ...MOCK_UNISSUED_FACILITY.details, bankReviewDate: new Date().toISOString() } };
+
+      const mapSummaryParams = {
+        app: MOCK_AIN_APPLICATION,
+        user: MOCK_REQUEST,
+      };
+
+      const actionItems = mapSummaryList(facility, mockedDisplayItemsEnd, mapSummaryParams, true)[0].actions.items;
+      // should be allowed to change so should display change
+      expect(actionItems).toEqual([
+        {
+          attributes: {
+            'data-cy': `bankReviewDate-action`,
+          },
+          classes: 'govuk-!-display-none',
+        },
+      ]);
+    });
+
+    it('facilityEndDate', () => {
+      const mockedDisplayItemsEnd = [
+        {
+          label: 'Facility end date',
+          id: 'facilityEndDate',
+        },
+      ];
+
+      const facility = { ...MOCK_UNISSUED_FACILITY, details: { ...MOCK_UNISSUED_FACILITY.details, facilityEndDate: new Date().toISOString() } };
+
+      const mapSummaryParams = {
+        app: MOCK_AIN_APPLICATION,
+        user: MOCK_REQUEST,
+      };
+
```

Comment:
For these tests it looks like they're all pretty similar between coverstartdate, coverenddate, is using FED, FED and bank review - what do you think about doing something like:


```
  const rowsToTest = [
    { id: 'coverStartDate', 
      label: 'Cover Start date', 
      facility: MOCK_UNISSUED_FACILITY 
    },
    {
      id: 'isUsingFacilityEndDate',
      label: 'Is using facility end date',
      facility: { ...MOCK_UNISSUED_FACILITY, details: { ...MOCK_UNISSUED_FACILITY.details, isUsingFacilityEndDate: true } },
    },
......
  ];
```

and then doing an `it.each` for all the tests below?

Reply:
yeahhhh I've been psyching myself up to improve this file